sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -599,10 +599,8 @@ void CurlSession::ReadStatusLineAndHeadersFromRawResponse(
 }
 
 // Read from curl session
-int64_t CurlSession::Read(Context const& context, uint8_t* buffer, int64_t count)
+int64_t CurlSession::ReadImplementation(Context const& context, uint8_t* buffer, int64_t count)
 {
-  context.ThrowIfCanceled();
```

Comment:
We should keep and move this cancel into the inner loop and before each call to `ReadFromSocket`, so that we don't wait more than one I/O operation to honor the cancelation.

Reply:
TransportPolicy checks for cancellation. This is the `don't wait more than one I/O operation honoring`.

Then `ReadFromSocket` checks for cancellation if the I/O can't be done when requested and until the I/O starts 🔥 !